x-pack/test/cloud_security_posture_functional/pages/custom_role_access.ts

```diff
@@ -0,0 +1,310 @@
+        name: 'some lower case rule name',
+        section: 'another lower case section',
+        benchmark: {
+          id: 'cis_gcp',
+          posture_type: 'cspm',
+          name: 'CIS Google Cloud Platform Foundation',
+          version: 'v2.0.0',
+        },
+        type: 'process',
+      },
+    },
+  ];
+
+  const VALID_CUSTOM_ROLE = {
+    name: 'valid_role',
+    elasticsearch: {
+      cluster: [],
+      indices: [
+        {
+          names: ['log-cloud_security_posture.findings-*'],
+          privileges: ['read'],
+        },
+        {
+          names: ['logs-cloud_security_posture.findings_latest-*'],
+          privileges: ['read'],
+        },
+        {
+          names: ['logs-cloud_security_posture.scores-*'],
+          privileges: ['read'],
+        },
+      ],
+      run_as: [],
+    },
+
+    kibana: [
+      {
+        base: ['all'],
+        spaces: ['*'],
+      },
+    ],
+  };
+
+  const INVALID_CUSTOM_ROLE = {
+    name: 'invalid_role',
+    elasticsearch: {
+      cluster: [],
+      indices: [
+        {
+          names: ['log-cloud_security_posture.findings-*'],
+          privileges: ['read'],
+        },
+        {
+          names: ['logs-cloud_security_posture.scores-*'],
+          privileges: ['read'],
+        },
+      ],
+      run_as: [],
+    },
+
+    kibana: [
+      {
+        base: ['all'],
+        spaces: ['*'],
+      },
+    ],
+  };
+
+  const VALID_USER = {
+    name: 'csp_valid_user',
+    password: 'test123',
+  };
+  const INVALID_USER = {
+    name: 'csp_invalid_user',
+    password: 'test123',
+  };
+
+  const navigateToHomePage = async () => {
+    await pageObjects.common.navigateToUrl('app');
+    await pageObjects.header.waitUntilLoadingHasFinished();
+  };
+
+  describe('Custom Role Access', function () {
+    let findings: typeof pageObjects.findings;
+    let latestFindingsTable: typeof findings.latestFindingsTable;
+    let cspDashboard: typeof pageObjects.cloudPostureDashboard;
+    let dashboard: typeof pageObjects.cloudPostureDashboard.dashboard;
+    let rule: typeof pageObjects.rule;
+    let benchmark: typeof pageObjects.benchmark;
+
+    before(async () => {
+      findings = pageObjects.findings;
+      latestFindingsTable = findings.latestFindingsTable;
+      cspDashboard = pageObjects.cloudPostureDashboard;
+      dashboard = pageObjects.cloudPostureDashboard.dashboard;
+      benchmark = pageObjects.benchmark;
+      rule = pageObjects.rule;
+
+      await kibanaServer.savedObjects.clean({
+        types: ['cloud-security-posture-settings'],
+      });
+
+      await findings.waitForPluginInitialized();
+
+      // Prepare mocked findings
+      await findings.index.remove();
+      await findings.index.add(data);
+    });
+
+    after(async () => {
+      await findings.index.remove();
+    });
+
+    describe('Access with custom roles', async () => {
+      before(async () => {
+        await security.role.create(VALID_CUSTOM_ROLE.name, {
```

Comment:
fyi, this can be initialized as part of kibana configuration. in the tests config file

Reply:
I think it would be better for future developers editing this code to be explicit and create the users and roles during the test.

Reply:
I think it better to have it explicit during the test.

Reply:
make sure to follow if there's a performance hit doing it here vs setting it up once through the configuration

Reply:
I agree, it now [creates](https://github.com/elastic/kibana/pull/184622/files#diff-7863c190bc746eb97a0db2090dd0c48915e91731811688621e5f48f68115f098R14) users and roles only once.